File: Data Structures & Algorithms/house-robber/submission-0.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int N = nums.size();

        if (N == 1) return nums[0];

        int a = nums[0];
        int b = max(nums[0], nums[1]);
        int c = b;

        for (int i = 2; i < N; i++) {
            c = max(a + nums[i], b);
            a = b;
            b = c;
        }

        return c;
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        arr=[]
        a=False
        for i in range(len(nums)):
            if nums[i] not in(arr):
                arr.append(nums[i])         
            else:
                a= True
                break
        if a==True:
            return True
        else:
            return False


